lib/improver/ensemble_copula_coupling/ensemble_copula_coupling.py

```diff
@@ -353,23 +360,42 @@ def _add_bounds_to_thresholds_and_probabilities(
                     each end.
         """
         lower_bound, upper_bound = bounds_pairing
-        threshold_points = insert_lower_and_upper_endpoint_to_1d_array(
-            threshold_points, lower_bound, upper_bound)
+        threshold_points_with_endpoints = \
+            insert_lower_and_upper_endpoint_to_1d_array(
+                threshold_points, lower_bound, upper_bound)
         probabilities_for_cdf = concatenate_2d_array_with_2d_array_endpoints(
             probabilities_for_cdf, 0, 1)
-        if np.any(np.diff(threshold_points) < 0):
-            msg = ("The end points added to the threshold values for "
-                   "constructing the Cumulative Distribution Function (CDF) "
-                   "must result in an ascending order. "
-                   "In this case, the threshold points {} must be "
-                   "outside the allowable range given by the "
-                   "bounds {}".format(
-                       threshold_points, bounds_pairing))
-            raise ValueError(msg)
-        return threshold_points, probabilities_for_cdf
+
+        if np.any(np.diff(threshold_points_with_endpoints) < 0):
+            msg = ("The calculated threshold values {} are not in ascending "
+                   "order as required for the cumulative distribution "
+                   "function (CDF). This is due to the threshold values "
+                   "exceeding the range given by the ECC bounds {}."
+                   .format(threshold_points_with_endpoints, bounds_pairing))
+            # If ecc_bounds_warning has been set, generate a warning message
+            # rather than raising an exception so that subsequent processing
+            # can continue. Then apply the new bounds as necessary to
+            # ensure the threshold values and endpoints are in ascending
+            # order and avoid problems further along the processing chain.
+            if ecc_bounds_warning:
+                warn_msg = msg + ("The threshold points that have exceeded "
```

Comment:
I think we need to insert either a space or a new-line character "\n" at the start of the added string.

Reply:
Required space added at beginning of string.